org.adempiere.base/src/org/compiere/model/MDepositBatch.java

```diff
@@ -344,5 +397,248 @@ public boolean isComplete()
+			setDocAction(DOCACTION_Complete);
+		
+		return DocAction.STATUS_InProgress;
+	}
+
+
+	@Override
+	public boolean approveIt() {
+		// TODO Auto-generated method stub
+		return false;
+	}
+
+
+	@Override
+	public boolean rejectIt() {
+		// TODO Auto-generated method stub
+		return false;
+	}
+
+
+	@Override
+	public String completeIt() {
+		//	Re-Check
+		if (!m_justPrepared)
+		{
+			String status = prepareIt();
+			m_justPrepared = false;
+			if (!DocAction.STATUS_InProgress.equals(status))
+				return status;
+		}
+		
+		m_processMsg = ModelValidationEngine.get().fireDocValidate(this, ModelValidator.TIMING_BEFORE_COMPLETE);
+		if (m_processMsg != null)
+			return DocAction.STATUS_Invalid;
+		
+		String sql = "SELECT COUNT(DISTINCT C_Currency_ID) FROM C_Payment p "
+				+ "INNER JOIN C_DepositBatchLine dbl ON dbl.C_Payment_ID = p.C_Payment_ID "
+				+ "WHERE dbl.C_DepositBatch_ID = ? ";
+		
+		int currencyCount = DB.getSQLValue (get_TrxName(), sql, getC_DepositBatch_ID());
+		if (currencyCount > 1)
+		{
+			m_processMsg = Msg.getMsg(getCtx(), "ErrorMultipleCurrencyPaymentsRestricted", new Object[] { getC_Currency().getISO_Code()} ); 
+			return DocAction.STATUS_Invalid;
+		}
+		
+		if (log.isLoggable(Level.INFO)) log.info("completeIt - " + toString());
+		
+		MDepositBatchLine[] depositbatchLines = getLines();
+		//	Close lines
+		for (int line = 0; line < depositbatchLines.length; line++)
+		{
+			depositbatchLines[line].setProcessed(true);
+			depositbatchLines[line].saveEx();
+		}
+		
+		//Re-calculate lines total amount and Update Header Deposit Amount
+		updateHeaderAmt();
+		
+		//	User Validation
+		String valid = ModelValidationEngine.get().fireDocValidate(this, ModelValidator.TIMING_AFTER_COMPLETE);
+		if (valid != null)
+		{
+			m_processMsg = valid;
+			return DocAction.STATUS_Invalid;
+		}
+			
+		setProcessed(true);
+		
+		setDocAction(DOCACTION_Close);
+		return DocAction.STATUS_Completed;
+	}
+
+
+	@Override
+	public boolean closeIt() {
+		if (log.isLoggable(Level.INFO)) log.info("closeIt - " + toString());
+		// Before Close
+		m_processMsg = ModelValidationEngine.get().fireDocValidate(this,ModelValidator.TIMING_BEFORE_CLOSE);
+		if (m_processMsg != null)
+			return false;		
+
+		setDocAction(DOCACTION_None);
+
+		// After Close
+		m_processMsg = ModelValidationEngine.get().fireDocValidate(this,ModelValidator.TIMING_AFTER_CLOSE);
+		if (m_processMsg != null)
+			return false;
+		return true;
+	}
+
+
+	@Override
+	public boolean reverseCorrectIt() {
+		if (log.isLoggable(Level.INFO)) log.info("reverseCorrectIt - " + toString());
+		// Before reverseCorrect
+		m_processMsg = ModelValidationEngine.get().fireDocValidate(this,ModelValidator.TIMING_BEFORE_REVERSECORRECT);
+		if (m_processMsg != null)
+			return false;
+		
+		// After reverseCorrect
+		m_processMsg = ModelValidationEngine.get().fireDocValidate(this,ModelValidator.TIMING_AFTER_REVERSECORRECT);
+		if (m_processMsg != null)
+			return false;
+		return false;

```

Comment:
the two modelvalidationengine call above is not needed since this is not implemented and always return false.